jupyter_server_fileid/manager.py

```diff
@@ -254,6 +242,45 @@ def __init__(self, *args, **kwargs):
         self.con.execute("CREATE INDEX IF NOT EXISTS ix_Files_path ON Files (path)")
         self.con.commit()
 
+    @staticmethod
+    def _normalize_separators(path):
+        """Replaces backslashes with forward slashes, removing adjacent slashes."""
+
+        parts = path.strip("\\").split("\\")
+        return "/".join(parts)
+
+    def _normalize_path(self, path):
+        """Accepts an API path and returns a "persistable" path, i.e. one prefixed
+        by root_dir that can then be persisted in a format relative to the implementation."""
+        # use commonprefix instead of commonpath, since root_dir may not be a
+        # absolute POSIX path.
+
+        norm_root_dir = self._normalize_separators(self.root_dir)
+        if posixpath.commonprefix([norm_root_dir, path]) != norm_root_dir:
+            path = posixpath.join(norm_root_dir, path)
+
+        path = self._normalize_separators(path)
```

Comment:
I think this method would be more clear if you moved this line before the `if` block above.

Reply:
ok.